File: pset6/sentimental-readability/readability.py
from cs50 import get_string
import re

# Prompt for user
text = get_string("Text: ").lower()

# Length of letters, words and sentences
letters = 0
for i in text:
    if i.isalpha():
        letters += 1
        
words = len(text.split())
sentences = len(re.findall(r'[\.\!\?]', text))

# Calculate Coleman-Liau formula
index = 0.0588 * letters / words * 100 - 0.296 * sentences / words * 100 - 15.8

# Print grade level
grade = round(index)
if grade <= 0:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    // Pega o texto do usuário
    string text = get_string("Text: ");

    // Calcula a quantidade de letras, palavras e sentenças
    int L = 0;
    int words = 0;
    int S = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (islower(text[i]) || isupper(text[i]))
        {
            L++;
        }
        else if (isspace(text[i]))
        {
            words++;
        }
        else if (text[i] == '!' || text[i] == '.' || text [i] == '?')
        {
            S++;
        }
    }
    words += 1;

    //Coleman-Liau Formula
    double index = 0.0588 * L / words * 100 - 0.296 * S / words * 100 - 15.8;
    int grade = (int) round(index);

    // Condição para saber o nível de escolaridade
    if (grade <= 0)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}